schema/types/labels.go

```diff
@@ -132,3 +143,64 @@ func LabelsFromMap(labelsMap map[ACIdentifier]string) (Labels, error) {
 	}
 	return labels, nil
 }
+
+// ToAppcOSArch translates a Golang arch tuple (OS, architecture, flavor) into
+// an appc arch tuple (OS, architecture)
+func ToAppcOSArch(goOs string, goArch string, goArchFlavor string) (appcOs string, appcArch string, e error) {
+	tabularAppcToGo := map[goArchTuple]appcArchTuple{
+		{"linux", "amd64", ""}:   {"linux", "amd64"},
+		{"linux", "386", ""}:     {"linux", "i386"},
+		{"linux", "arm64", ""}:   {"linux", "aarch64"},
+		{"linux", "arm", ""}:     {"linux", "armv6l"},
+		{"linux", "arm", "6"}:    {"linux", "armv6l"},
+		{"linux", "arm", "7"}:    {"linux", "armv7l"},
+		{"linux", "ppc64", ""}:   {"linux", "ppc64"},
+		{"linux", "ppc64le", ""}: {"linux", "ppc64le"},
+		{"linux", "s390x", ""}:   {"linux", "s390x"},
+
+		{"freebsd", "amd64", ""}: {"freebsd", "amd64"},
+		{"freebsd", "386", ""}:   {"freebsd", "i386"},
+		{"freebsd", "arm", ""}:   {"freebsd", "arm"},
+		{"freebsd", "arm", "5"}:  {"freebsd", "arm"},
+		{"freebsd", "arm", "6"}:  {"freebsd", "arm"},
+		{"freebsd", "arm", "7"}:  {"freebsd", "arm"},
+
+		{"darwin", "amd64", ""}: {"darwin", "x86_64"},
+		{"darwin", "386", ""}:   {"darwin", "i386"},
+	}
+	archTuple, ok := tabularAppcToGo[goArchTuple{goOs, goArch, goArchFlavor}]
+	if !ok {
+		return "", "", fmt.Errorf("unknown arch tuple: %q - %q - %q", goOs, goArch, goArchFlavor)
+	}
+	return archTuple.appcOs, archTuple.appcArch, nil
+}
+
+// ToGoOSArch translates an appc arch tuple (OS, architecture) into
+// a Golang arch tuple (OS, architecture, flavor)
+func ToGoOSArch(appcOs string, appcArch string) (goOs string, goArch string, goArchFlavor string, e error) {
+	tabularGoToAppc := map[appcArchTuple]goArchTuple{
+		// {"linux", "aarch64_be"}: nil,
+		// {"linux", "armv7b"}: nil,
```

Comment:
We can, this is here just to make explicit that there are two arch without a Go equivalent (future-proofing myself the next time I go over the list). I'd better leave this here or translate to a human-readable comment.